src/main/java/it/pagopa/ecommerce/commons/domain/TransactionClosed.java

```diff
@@ -52,10 +54,30 @@ public TransactionClosed withStatus(TransactionStatusDto status) {
                         new TransactionWithRequestedAuthorization(
                                 new TransactionActivated(
                                         this.getTransactionId(),
-                                        new PaymentToken(this.getTransactionActivatedData().getPaymentToken()),
-                                        this.getRptId(),
-                                        this.getDescription(),
-                                        this.getAmount(),
+                                        this.getNoticeCodes().stream()
+                                                .filter(
+                                                        noticeCode -> this.getTransactionActivatedData()
+                                                                .getNoticeCodes().stream().map(n -> n.getRptId())
+                                                                .collect(Collectors.toList())
+                                                                .contains(noticeCode.rptId().value())
```

Comment:
Is there any difference between the notice codes present in the non-activated transaction and the ones in the activated one? If not this could be simplified, as the lamdba inside the filter would always return true

Reply:
We can trust that they are the same, so we can delete the filter expression